core/env/replace_spec.rb

```diff
@@ -1,15 +1,52 @@
 require_relative '../../spec_helper'
 
 describe "ENV.replace" do
+  before :each do
+    @orig = ENV.to_hash
+    ENV.delete("foo")
+  end
+
+  after :each do
+    ENV.replace(@orig)
+  end
 
   it "replaces ENV with a Hash" do
-    ENV["foo"] = "bar"
-    e = ENV.reject { |k, v| k == "foo" }
-    e["baz"] = "bam"
-    ENV.replace e
-    ENV["foo"].should == nil
-    ENV["baz"].should == "bam"
-    ENV.delete "baz"
+    ENV.replace("foo" => "0", "bar" => "1").should equal(ENV)
+    ENV.size.should == 2
+    ENV["foo"].should == "0"
+    ENV["bar"].should == "1"
+  end
+
+  it "raises TypeError if the argument is not a Hash" do
+    -> { ENV.replace("foo" => "0", "bar" => "1").should raise_error(TypeError, "no implicit conversion of Object into Hash") }
+    ENV.to_hash.should == @orig
+  end
+
+  it "raises TypeError if a key is not a String" do
+    -> { ENV.replace(Object.new => "0").should raise_error(TypeError, "no implicit conversion of Object into String") }
```

Comment:
Same here and below, the `.should raise_error` should be after the `}`.
I wonder why the CI doesn't fail though.

Reply:
Ah, it's because this just create a big lambda literal, but actually never calls it.
And the spec still "pass" because of the expectation below which passes.

Reply:
K